### PR TITLE
Generify functions in document.rs

### PR DESCRIPTION
**Pull request by TrueDoctor**
image

Closes: Nothing

### CODE REVIEW

File: graphene/src/document.rs [hunk 1]
```diff
@@ -98,8 +98,11 @@ impl Document {
 		layers.reduce(|a, b| &a[..a.iter().zip(b.iter()).take_while(|&(a, b)| a == b).count()]).unwrap_or_default()
 	}
 
-	pub fn folders<'a>(&'a self, layers: impl Iterator<Item = &'a [LayerId]>) -> impl Iterator<Item = &'a [LayerId]> {
-		layers.filter(|layer| self.is_folder(layer))
+	pub fn folders<'a, T: AsRef<[LayerId]>>(&'a self, layers: impl Iterator<Item = T> + 'a) -> impl Iterator<Item = T> + 'a
```
Comment (asyncth) [inline graphene/src/document.rs:101]: `T: AsRef<[LayerId]>` probably should be omitted because it's already in `where` clause.

Comment (TrueDoctor) [reply graphene/src/document.rs:101]: yeah, sorry forgot that

**Comment by Keavon** [comment]
I approve (wholeheartedly) of the meme. But I haven't given my approval of the code since I lack the context about what this changes or solves. I'd rather defer to others who are better equipped to evaluate this since I trust the community better than me here.

**Comment by TrueDoctor** [comment]
@Keavon sorry about that, meant to add doc comments to the function to explain what they are doing, just didn't get around to it yet

**Review by Keavon — APPROVED** [review]

**Pull request merged**